app/src/main/java/de/westnordost/streetcomplete/ApplicationModule.kt

```diff
@@ -15,4 +16,5 @@ val appModule = module {
     single { CrashReportExceptionHandler(androidContext(), get(), "streetcomplete_errors@westnordost.de", "crashreport.txt") }
     single { DatabaseLogger(get()) }
     single { SoundFx(androidContext()) }
+    single { HttpClient() }
```

Comment:
Is HttpClient threadsafe? I.e. is it recommended to use one instance of HttpClient for everything, or create new instances for different addresses?

In any case, the module is the perfect place to set a few settings on the HttpClient. Is it for example possible to set the default user agent here to be added to the header of every request?

Reply:
From what I've read, the [HttpClient creates its own threads](https://ktor.io/docs/create-client.html#close-client), so based on that I'd assume that it is threadsafe. The guidance also recommend:

> Note that creating HttpClient is not a cheap operation, and it's better to reuse its instance in the case of multiple requests.